### ISSUE TITLE
RabbitMQ fails to add user

### ISSUE THREAD

**Issue by kuddl** [puphpet/puphpet#1470]
Hi there, 
Thanks a lot for creating Puphpet! It's a great tool!

I tried to use the new feature of adding a a User to RabbitMQ.
But it failed ....

Here is the error:
``` 
==> default: Notice: /Stage[main]/Rabbitmq::Service/Service[rabbitmq-server]: Triggered 'refresh' from 1 events
==> default: Error: Could not set 'present' on ensure: undefined method `member?' for nil:NilClass
==> default: Error: Could not set 'present' on ensure: undefined method `member?' for nil:NilClass
==> default: Wrapped exception:
==> default: undefined method `member?' for nil:NilClass
==> default: Error: /Stage[main]/Main/Rabbitmq_user[tobias]/ensure: change from absent to present failed: Could not set 'present' on ensure: undefined method `member?' for nil:NilClass
``` 

Config.yaml
```
---
vagrantfile-local:
    vm:
        box: puphpet/debian75-x64
        box_url: puphpet/debian75-x64
        hostname: ''
        memory: '1024'
        cpus: '2'
        chosen_provider: virtualbox
        network:
            private_network: 192.168.56.102
            forwarded_port:
                OqHduSPHo3JS:
                    host: '8019'
                    guest: '22'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: on
            vmware:
                numvcpus: 1
            parallels:
                cpus: 1
        provision:
            puppet:
                manifests_path: puphpet/puppet
                manifest_file: site.pp
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
                    - '--parser future'
        synced_folder:
            pDpUiiTVu9qH:
                owner: www-data
                group: www-data
                source: ./
                target: /var/www
                sync_type: default
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                    auto: 'false'
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
    vagrant:
        host: detect
server:
    install: '1'
    packages:
        - vim
        - htop
        - ack-grep
users_groups:
    install: '1'
    groups: {  }
    users: {  }
cron:
    install: '1'
    jobs: {  }
firewall:
    install: '1'
    rules: {  }
apache:
    install: '1'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        pma1qvgg567t:
            servername: awesome.dev
            serveraliases:
                - www.awesome.dev
            docroot: /var/www/awesome
            port: '80'
            setenv:
                - 'APP_ENV dev'
            custom_fragment: ''
            directories:
                pvd7lq4w33xp:
                    provider: directory
                    path: /var/www/awesome
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        qwvaklmcgeoj:
                            provider: filesmatch
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
    upstreams: {  }
    vhosts:
        4aimciw1ruah:
            proxy: ''
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            locations:
                k04pbmesmvy1:
                    location: /
                    try_files:
                        - $uri
                        - $uri/
                        - index.php
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                re4bkktpvapf:
                    location: '~ \.php$'
                    try_files:
                        - $uri
                        - $uri/
                        - index.php
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
php:
    install: '1'
    settings:
        version: '56'
    fpm_settings:
        port: 9000
    composer: '1'
    composer_home: ''
    modules:
        php:
            - cli
            - intl
            - mcrypt
        pear: {  }
        pecl:
            - pecl_http
    ini:
        display_errors: On
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings:
        host: 127.0.0.1
        port: '9000'
    ini:
        display_errors: On
        error_reporting: '-1'
    timezone: null
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.17.1
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions: {  }
nodejs:
    install: '1'
    npm_packages:
        - hapi
        - gulp
        - pageres
python:
    install: '1'
    packages: {  }
    versions: {  }
mysql:
    install: '1'
    settings:
        version: '5.6'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        IcjK9yRzGjlH:
            name: dbuser
            password: '123'
    databases:
        JJYHIgJtw3X7:
            name: dbname
            sql: ''
    grants:
        ozLzpYVv5wa5:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.3'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mongodb:
    install: '0'
    settings:
        auth: 1
        bind_ip: 127.0.0.1
        port: '27017'
    databases: {  }
redis:
    install: '1'
    settings:
        conf_port: '6379'
mailcatcher:
    install: '1'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '1080'
        mailcatcher_path: /usr/local/rvm/wrappers/default
        from_email_method: inline
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '13000'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
    binlogdir: /var/lib/beanstalkd/binlog
rabbitmq:
    install: '1'
    settings:
        port: '5672'
    users:
        etprdvaogrpx:
            name: tobias
            password: tobias
            tags:
                - administrator
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 1.4.1
        java_install: true
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'


```

**Comment by fracasula** [comment]
Same here

```yaml
rabbitmq:
    install: '1'
    settings:
        port: '5672'
        management_port: '15672'
        admin_enable: true
        service_manage: true
    users:
        rmqu_oko0n1y8vvqo:
            admin: true
            name: vagrant
            password: password
            permissions:
                rmqup_8olnvka9re3r:
                    host: symfony_dev
                    configure_permission: '.*'
                    read_permission: '.*'
                    write_permission: '.*'
                rmqup_fvh9ttmbyhak:
                    host: symfony_test
                    configure_permission: '.*'
                    read_permission: '.*'
                    write_permission: '.*'
    vhosts:
        - symfony_dev
        - symfony_test
    plugins: {}
```

```sh
==> default: Notice: /Stage[main]/Rabbitmq::Install/Package[rabbitmq-server]/ensure: ensure changed 'purged' to 'present'
==> default: Info: /Stage[main]/Rabbitmq::Install/Package[rabbitmq-server]: Scheduling refresh of Class[Rabbitmq::Service]
==> default: Notice: /Stage[main]/Rabbitmq/Rabbitmq_plugin[rabbitmq_management]/ensure: created
==> default: Info: /Stage[main]/Rabbitmq/Rabbitmq_plugin[rabbitmq_management]: Scheduling refresh of Class[Rabbitmq::Service]
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[rabbitmq.config]/ensure: defined content as '{md5}5647446b04534bab24f4b348af808e93'
==> default: Info: /Stage[main]/Rabbitmq::Config/File[rabbitmq.config]: Scheduling refresh of Class[Rabbitmq::Service]
==> default: Notice: /Stage[main]/Main/Puphpet::Php::Module[mysql]/Php::Module[mysql]/Package[PhpModule_mysql]/ensure: created
==> default: Info: /Stage[main]/Main/Puphpet::Php::Module[mysql]/Php::Module[mysql]/Package[PhpModule_mysql]: Scheduling refresh of Service[php5-fpm]
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[/etc/rabbitmq/ssl]/ensure: created
==> default: Notice: /Stage[main]/Pyenv/Vcsrepo[/usr/local/pyenv]/revision: revision changed '3cde540ca00208fbfd0a5e7708c3053bc8cca9f5' to 'v0.4.0-20140404'
==> default: Notice: /Stage[main]/Composer/Exec[download_composer]/returns: executed successfully
==> default: Notice: /Stage[main]/Rabbitmq::Config/File[rabbitmq-env.config]/ensure: defined content as '{md5}6513cd3ef9805aa61b060ae020df4fd5'
==> default: Info: /Stage[main]/Rabbitmq::Config/File[rabbitmq-env.config]: Scheduling refresh of Class[Rabbitmq::Service]
==> default: Info: Class[Rabbitmq::Config]: Scheduling refresh of Class[Rabbitmq::Service]
==> default: Info: Class[Rabbitmq::Service]: Scheduling refresh of Service[rabbitmq-server]
==> default: Notice: /Stage[main]/Rabbitmq::Service/Service[rabbitmq-server]: Triggered 'refresh' from 1 events
==> default: Error: Could not set 'present' on ensure: undefined method `add' for nil:NilClass
==> default: Error: Could not set 'present' on ensure: undefined method `add' for nil:NilClass
==> default: Wrapped exception:
==> default: undefined method `add' for nil:NilClass
==> default: Error: /Stage[main]/Main/Rabbitmq_user[vagrant]/ensure: change from absent to present failed: Could not set 'present' on ensure: undefined method `add' for nil:NilClass
==> default: Notice: /Stage[main]/Rabbitmq::Management/Rabbitmq_user[guest]/ensure: removed
==> default: Notice: /Stage[main]/Rabbitmq::Install::Rabbitmqadmin/Staging::File[rabbitmqadmin]/Exec[/var/lib/rabbitmq/rabbitmqadmin]/returns: executed successfully
==> default: Notice: /Stage[main]/Rabbitmq::Install::Rabbitmqadmin/File[/usr/local/bin/rabbitmqadmin]/ensure: defined content as '{md5}05e0d35eda1d72cebbd7e502ea33736a'
==> default: Notice: /Stage[main]/Main/Puphpet::Php::Pecl[pecl_http]/Php::Pecl::Module[pecl_http]/Exec[pecl-pecl_http]/returns: executed successfully
==> default: Notice: /Stage[main]/Puphpet::Php::Xdebug/Vcsrepo[/.puphpet-stuff/xdebug]/revision: revision changed '8737484ca6cc8f05ab095d7249952fcb5955fe57' to 'XDEBUG_2_3_1'
==> default: Notice: /Stage[main]/Main/Service[php5-fpm]/ensure: ensure changed 'stopped' to 'running'
==> default: Info: /Stage[main]/Main/Service[php5-fpm]: Unscheduling refresh on Service[php5-fpm]
==> default: Notice: /Stage[main]/Main/Puphpet::Php::Pecl[amqp]/Php::Pecl::Module[amqp]/Exec[pecl-amqp]/returns: executed successfully
==> default: Notice: /Stage[main]/Main/File[/var/run/apache2/ssl_mutex]/ensure: created
==> default: Info: /Stage[main]/Main/File[/var/run/apache2/ssl_mutex]: Scheduling refresh of Service[httpd]
==> default: Notice: /Stage[main]/Apache::Service/Service[httpd]: Triggered 'refresh' from 1 events
==> default: Notice: /Stage[main]/Main/Rabbitmq_vhost[symfony_dev]/ensure: created
==> default: Notice: /Stage[main]/Main/Rabbitmq_vhost[symfony_test]/ensure: created
==> default: Notice: /Stage[main]/Main/Rabbitmq_user_permissions[vagrant@symfony_test]: Dependency Rabbitmq_user[vagrant] has failures: true
==> default: Warning: /Stage[main]/Main/Rabbitmq_user_permissions[vagrant@symfony_test]: Skipping because of failed dependencies
==> default: Notice: /Stage[main]/Main/Rabbitmq_user_permissions[vagrant@symfony_dev]: Dependency Rabbitmq_user[vagrant] has failures: true
==> default: Warning: /Stage[main]/Main/Rabbitmq_user_permissions[vagrant@symfony_dev]: Skipping because of failed dependencies
==> default: Notice: Finished catalog run in 76.99 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Thanks for creating puphpet anyway, great stuff :+1:

**Comment by jtreminio** [comment]
Fixed by 52ea231
